src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -574,19 +613,41 @@ public void initialize(HttpRequest request)
                             request.setParser(new JsonObjectParser(JSON_FACTORY));
                         }
                     });
-            GenericUrl url = new GenericUrl(provider.authorization_endpoint);
             HttpRequest request = requestFactory.buildGetRequest(url);
-            request.execute().getStatusCode();
-        } catch (com.google.api.client.http.HttpResponseException hre) {
-            if (hre.getStatusCode() == 404) {
-                LOGGER.log(Level.FINE, "test auth endpoint", hre);
+            com.google.api.client.http.HttpResponse response = request.execute();
+            int rc = response.getStatusCode();
+            if (rc != 200)
+                LOGGER.info("OpenShift OAuth the attempt to get the server version request got an unexpected return code: " + rc);
+            OpenShiftVersionInfo version = response.parseAs(OpenShiftVersionInfo.class);
+            if (version != null && version.major != null && version.major.equals("1")) {
+                if (version.minor.length() > 2) {
+                    String minor = version.minor.substring(0, 2); // per javadoc end index is not inclusive
+                    int m = Integer.parseInt(minor);
+                    if (m <= 11) {
+                        // 3.x cluster
+                        LOGGER.info("OpenShift OAuth the server is 3.x, specifically " + version.toString());
```

Comment:
Are the version numbers for the OpenShift OAuth server completely independent from the product version?

Reply:
What is being checked here is the underlying kubernetes version. It is effectively independent with 4.0.

Reply:
Yep it is just a debug marker to note which path wrt transports, oauth endpoints, and certs is being taken

OAuth behavior is different between 3.x and 4.x @adambkaplan , so we vary accordingly.  Goal is to be able to use 1 version of the login plugin across 3.x and 4.x to simplify support.